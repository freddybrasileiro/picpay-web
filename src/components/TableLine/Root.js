import styled from "styled-components";

export default styled.section`
  padding: 10px 0px 2px 0px;
  border-bottom: solid 0.5px #d8d8d8;
  display: flex;
  min-width: 140px;

  p {
    color: ${props => (props.theme.width === "xs" ? "white" : "#474971")};

    :first-child {
      flex: 1;
    }
    :last-child {
      font-weight: bold;
      color: #7de6c6;
    }
  }
`;
