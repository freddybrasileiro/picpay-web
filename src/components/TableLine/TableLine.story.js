import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { IntlProvider } from "react-intl";

import styledComponentsTheme from "styledComponentsTheme";
import { DEFAULT_LOCALE, translationMessages } from "i18n";

import TableLine from "components/TableLine";

storiesOf("Components/TableLine", module)
  .addDecorator(story => (
    <IntlProvider
      locale={DEFAULT_LOCALE}
      messages={translationMessages[DEFAULT_LOCALE]}
    >
      <ThemeProvider theme={styledComponentsTheme}>
        <div style={{ maxWidth: 500, margin: "10px 10px" }}>{story()}</div>
      </ThemeProvider>
    </IntlProvider>
  ))
  .add("Default", () => (
    <TableLine data={{ label: "Transação", value: "8888888" }} />
  ));
