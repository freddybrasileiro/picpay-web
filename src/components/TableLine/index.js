import React from "react";
import PropTypes from "prop-types";
import { Typography, withWidth } from "@material-ui/core";

import Root from "./Root";

const Receipt = ({ data, width }) => {
  return (
    <Root theme={{ width }}>
      <Typography component="p" variant="headline">
        {data.label}
      </Typography>
      <Typography component="p" variant="headline">
        {data.value}
      </Typography>
    </Root>
  );
};

const propTypes = {
  data: PropTypes.object
};

const defaultProps = {
  data: {}
};

Receipt.propTypes = propTypes;
Receipt.defaultProps = defaultProps;

export default withWidth()(Receipt);
