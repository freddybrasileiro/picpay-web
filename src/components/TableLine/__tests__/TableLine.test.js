import React from "react";
import { createShallow } from "@material-ui/core/test-utils";

import TableLine from "components/TableLine";

describe("TableLine", () => {
  const shallow = createShallow({ dive: true });

  it("Should render the TableLine", () => {
    expect(
      shallow(<TableLine data={{ label: "Transação", value: "8888888" }} />)
    ).toMatchSnapshot();
  });
});
