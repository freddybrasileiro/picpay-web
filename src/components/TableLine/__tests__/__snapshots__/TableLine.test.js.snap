// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableLine Should render the TableLine 1`] = `
<Fragment>
  <Receipt
    data={
      Object {
        "label": "Transação",
        "value": "8888888",
      }
    }
    width="md"
  />
  <EventListener
    onResize={[Function]}
    target="window"
  />
</Fragment>
`;
